tests/test_utils.rs

```diff
@@ -37,3 +40,31 @@ pub fn continue_to_end(target: &LinuxTarget) {
         status => panic!("Status: {:?}", status),
     }
 }
+
+/// Turn the `pause` instruction inside the `breakpoint` function into a breakpoint.
+#[cfg(target_os = "linux")]
+#[allow(dead_code)]
+pub fn patch_breakpoint(target: &LinuxTarget, debuginfo: &RelocatedDwarf) {
+    // Test that `a_function` resolves to a function.
+    let breakpoint_addr = debuginfo.get_symbol_address("breakpoint").unwrap() + 4 /* prologue */;
+    // Write breakpoint to the `breakpoint` function.
+    let mut pause_inst = 0 as libc::c_ulong;
+    unsafe {
+        target
+            .read()
+            .read(&mut pause_inst, breakpoint_addr)
+            .apply()
+            .unwrap();
+    }
+    // pause (rep nop); ...
+    assert_eq!(&pause_inst.to_ne_bytes()[0..2], &[0xf3, 0x90]);
```

Comment:
Might be a good place to use Capstone or iced-x86 for instruction generation, but it is out of scope of this pull request, and it will go into a separate `Breakpoint` module anyway. I think we can add a TODO for future reference, though - e.g.:
```suggestion
    // TODO: use an instruction generator to convert assembly into binary code
    assert_eq!(&pause_inst.to_ne_bytes()[0..2], &[0xf3, 0x90]);
```